### DISCUSSION TITLE
Run onnxruntime in wasm runtime

### DISCUSSION THREAD

**Discussion by chris-void-777** [microsoft/onnxruntime#16849] [Other Q&A] [answered]
After reading the ONNX Runtime documentation, I learned that ONNX Runtime can run in the browser and can interact with JavaScript.Now, I want to run ONNX Runtime within a WebAssembly runtime embedded in C++. How can I interact with the code inside ort-wasm.wasm? Specifically, I want to konw how the JavaScript API calls the code inside WebAssembly. So that  I can write code to achieve the implementation I desire.
I have inspected the export section of ort-wasm.wasm:
```
wasm-tools print ort-wasm.wasm | grep export | head
  (export "N" (memory 0))
  (export "O" (func 2856))
  (export "P" (func 7114))
  (export "Q" (func 7343))
  (export "R" (func 7913))
  (export "S" (func 8130))
  (export "T" (func 8178))
  (export "U" (func 6918))
  (export "V" (func 7978))
  (export "W" (func 7005))
 ...
```
I don't know which functions to call.

Thanks for your time and assistance.

**Comment by xadupre** [comment, marked as answer]
I assume you read this page https://onnxruntime.ai/docs/tutorials/web/ but you did not find the information you need. Otherwise, the code you are looking for should be here: https://github.com/microsoft/onnxruntime/tree/main/js/web.

**Reply by Thisislegit** [reply]
Is there a c++ SDK for onnxruntime Web like in js?
